build-logic/convention-plugins/build.gradle.kts

```diff
@@ -2,15 +2,11 @@ plugins {
     `kotlin-dsl`
 }
 
-gradlePlugin {
-    plugins {
-        register("CodeQuality") {
-            id = "social.androiddev.code-quality"
-            implementationClass = "social.androiddev.plugins.CodeQualityPlugin"
-        }
-    }
-}
-
 dependencies {
     implementation(libs.com.diffplug.spotless.gradle.plugin)
+    implementation(libs.org.jetbrains.kotlin.gradle.plugin)
+    implementation(libs.org.jetbrains.compose.gradle.plugin)
+    implementation(libs.com.android.tools.build.gradle)
+    // hack to access version catalogue https://github.com/gradle/gradle/issues/15383
+    implementation(files(libs.javaClass.superclass.protectionDomain.codeSource.location))
```

Comment:
What about creating a version catalog in `settings.gradle.kts` - would it work instead of the hack?

```
    versionCatalogs {
        create("libs") {
            from(files(File(rootDir, "../gradle/libs.versions.toml")))
        }
    }
```

Reply:
That is already in settings.gradle.kts

Reply:
I missed that 🙈